### PR TITLE
docker-squash 1.0.6 (new formula)

### PR DESCRIPTION
**Pull request by kwilczynski**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by kwilczynski** [comment]
I run brew audit --strict --new-formula and it didn't found any issues. I also use this formula personally, since `docker build . --squash` does not sometimes work for me (not sure why, though), and this little utility works wonders, thus I decided to share with others.

**Comment by kwilczynski** [comment]
Hi @ilovezfs, I saw conversation in other Pull Requests about not using virtualenv approach, and I did try to do it in this case, but I couldn't get it to work correctly. I do apologise for inconvenience.

**Comment by kwilczynski** [comment]
Hi @ilovezfs, do you think this is good enough to be merged?

**Comment by kwilczynski** [comment]
Hi @DomT4, sorry to bother you, but I was wondering if there is anything wrong here that would prevent this formula to be added? I am happy to help as much as possible.

**Comment by DomT4** [comment]
Arguably both the dependencies should be `:recommended` rather than `:optional` but besides that I don't think there's anything particularly obvious that isn't a philosophical discussion around how to handle `python` stuff, so, this one comes down to @ilovezfs who has the power to merge/deny/etc. I'm a humble contributor alone these days 😉.

**Comment by kwilczynski** [comment]
@DomT4 thank you for getting back to me!

I left the `:recommended` out, as some people (myself included) often do not install Docker et al using Homebrew, and since recommended dependencies are installed automatically by default when missing, it seemed like it is best to keep these as optional.

It's all in @ilovezfs' hands then!

**Comment by kwilczynski** [comment]
Hi @ilovezfs, any change to get this new formula merged?

**Comment by vinyldarkscratch** [comment]
Pip isn't installed on the Brew test bot -- we should add an install step to download and install `pip` (I think).  See: https://pip.pypa.io/en/stable/installing/

**Comment by DomT4** [comment]
It should be added as a `resource`, although that this is necessary seems silly.

**Comment by commitay** [comment]
@BrewTestBot test this please

File: Formula/docker-squash.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+class DockerSquash < Formula
+  desc "Docker image squashing tool"
+  homepage "https://github.com/goldmann/docker-squash"
+  url "https://github.com/goldmann/docker-squash/archive/1.0.6.tar.gz"
+  sha256 "8430390edc2f19f368bacae57c17363bd64f76dbcd0cf00f4ec5ce6f8cbe9abd"
+
+  depends_on "python" if MacOS.version <= :snow_leopard
+
```
Comment (DomT4) [inline Formula/docker-squash.rb]: Don't need this empty line.

File: Formula/docker-squash.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  depends_on "docker"
+  depends_on "docker-machine"
+
+  resource "setuptools" do
+    url "https://files.pythonhosted.org/packages/a4/c8/9a7a47f683d54d83f648d37c3e180317f80dc126a304c45dc6663246233a/setuptools-36.5.0.zip"
+    sha256 "ce2007c1cea3359870b80657d634253a0765b0c7dc5a988d77ba803fc86f2c64"
+  end
+
+  # resource "docker" dependency
```
Comment (DomT4) [inline Formula/docker-squash.rb]: Not really sure what this comment is trying to say.

**Comment by kwilczynski** [comment]
Hi there. I wonder who is working on this formula? I am not doing anything to it, but I am getting a lot of notifications from GitHub. Anyone care to explain what is going on here? :)

**Comment by DomT4** [comment]
Both @ilovezfs & @commitay have been pushing tweaks to the formula. Maintainers have push access to opened PRs unless you explicitly disable their access to it, and semi-frequently use such access to push things over the line if something is 90% okay.

**Comment by kwilczynski** [comment]
Hi @DomT4, no issue with the access, etc., I was more thinking if there is anything I need to do? Especially, since you did put two comments recently. Given that @ilovezfs and @commitay are pushing updates, I wasn't sure if this was directed at me or them, or what is going on :) Apologies for commotion.

**Comment by DomT4** [comment]
>  I was more thinking if there is anything I need to do? 

If you wanted to pick this PR back up & address any remaining comments I'm sure ILZ/commitay would be happy for you to do so. A stronger test, if at all possible, would certainly be a useful addition.

**Review by ilovezfs — APPROVED** [review]

**Comment by commitay** [comment]
Thanks for the new formula @kwilczynski!

**Comment by kwilczynski** [comment]
Hi @ilovezfs, @commitay and @DomT4!

Thank you so much for help! It is true, indeed, that Homebrew community is the best! ❤️

**Pull request closed without merging**